### PR TITLE
revpi-eep: Add support for the 2nd gpio bank [REVPI-2792]

### PR DESCRIPTION
**Pull request by iluminat23**
This PR also includes the fix from @l1k https://github.com/RevolutionPi/revpi-hat-eeprom/pull/18

### CODE REVIEW

**Comment by ferdrich-kunbus** [comment]
Commit messages:

[4596934](https://github.com/RevolutionPi/revpi-hat-eeprom/pull/21/commits/459693467dbc241df8e8c37197589c662b58312e) : rellay -> really
[8a49938](https://github.com/RevolutionPi/revpi-hat-eeprom/pull/21/commits/8a499380fd66bac98c3f5daa7f035ac13cd09dc9) : futer -> future
[c8acb64](https://github.com/RevolutionPi/revpi-hat-eeprom/pull/21/commits/c8acb64fe7bc37b6cb5f5424a34070cd1e301ed7) : Implent -> Implement

File: revpi-hat-eep/src/gpio.rs [hunk 1]
```diff
@@ -6,7 +6,9 @@ use rpi_hat_eep::gpio_map;
 use serde::{Deserialize, Serialize};
 use std::fmt::Display;
 
-const MAX_GPIOS: usize = 28;
+const BANK0_GPIOS: usize = 28;
+const BANK1_GPIOS: usize = 18;
+const MAX_GPIOS: usize = BANK0_GPIOS + BANK1_GPIOS;
```
Comment (ferdrich-kunbus) [inline revpi-hat-eep/src/gpio.rs:11]: Duplicated constants, are alreadey defined in rpi-hat-eep/src/gpio.rs
Hard to maintain. Try to do something like this -> use rpi-hat-eep::BANK0_GPIOS or whatever is appropriate to remove duplication.

Some use constant-only files.

File: docs/eep.schema [hunk 2]
```diff
@@ -95,7 +95,7 @@
                                     "description": "GPIO or pin number",
                                     "type": "integer",
                                     "minimum": 2,
-                                    "maximum": 27
+                                    "maximum": 45
```
[the remark this answers is not in the document]
Comment (iluminat23) [reply docs/eep.schema:98]: Yeah, this is a json schema limitation. The schema is a nice to have but as json schemas are pretty limited this should not define how we model data. The final saying has the app not the schema.

File: revpi-hat-eep/src/bin/revpi-eep.rs [hunk 3]
```diff
@@ -97,9 +97,10 @@ fn create_rpi_eep(config: RevPiHatEeprom) -> Result<rpi_hat_eep::Eep, Box<dyn st
         config.pstr,
     )?;
 
-    let gpio_map: gpio_map::EepAtomGpioMapData = config.gpiobanks[0].clone().try_into()?;
-
-    let mut eep = Eep::new(vendor_data, gpio_map);
+    let gpio_bank0_map: gpio_map::EepAtomGpioMapData = config.gpiobanks[0]
+        .clone()
+        .into_gpio_map(gpio_map::GpioBank::Bank0)?;
+    let mut eep = Eep::new(vendor_data, gpio_bank0_map);
```
[the remark this answers is not in the document]
Comment (iluminat23) [reply revpi-hat-eep/src/bin/revpi-eep.rs:103]: The error is pretty easy to find.
1. The order should be clear to any body as there is no other way to describe the bank number as the order.
2. If you put a wrong gpio in the wrong bank the program will give you an pretty obvious error:  ```ERROR: Invalid config file `docs/example.json': gpio# (bank1): 3 MIN: 28, MAX: 45```

There are some braces missing in the error message. Should look like this:
```ERROR: Invalid config file `docs/example.json': gpio# (bank1): 3 (MIN: 28, MAX: 45)```

**Review by ferdrich-kunbus — APPROVED** [review]

**Pull request merged**